### PR TITLE
[refactor]: improve error messages on configuration failure.

### PR DESCRIPTION
**Pull request by appetrosyan**
Signed-off-by: Aleksandr Petrosyan <a-p-petrosyan@yandex.ru>

### Description of the Change

Improved error messages in cases where configuration failed to get parsed. 

### Issue

None

### Benefits

Better UX. 

### Possible Drawbacks

None

### CODE REVIEW

File: config/src/client.rs [hunk 1]
```diff
@@ -119,35 +123,36 @@ impl ConfigurationProxy {
...
-                    "`TORII_API_URL` string should provide a connection protocol".to_owned()
+                    "`TORII_API_URL` string: `{api_url}` should provide a connection protocol"
+                        .to_owned()
                 ));
             }
+            // TODO: this is neither robust, nor useful. This should be enforced as a `FromStr` implementation.
             if split_api_url[0] != "http" {
                 eyre::bail!(ConfigError::ProxyBuildError(
-                    "`TORII_API_URL` string only supports HTTP".to_owned()
+                    "`TORII_API_URL` string: `{api_url`} only supports the `HTTP` protocol currently".to_owned()
```
Comment (ilchu) [inline config/src/client.rs:155]:
```suggestion
                    "`TORII_API_URL` string: `{api_url}` only supports the `HTTP` protocol currently".to_owned()
```

File: config/src/client.rs [hunk 2]
```diff
@@ -107,6 +107,10 @@ impl Default for ConfigurationProxy {
     }
 }
 
+// TODO: explain why these values were chosen.
```
Comment (ilchu) [inline config/src/client.rs:110]: This was rather arbitrary tbh, just *unreasonably* small values I've drawn from my head. Maybe it invites further discussion.

File: config/src/client.rs (hunk 1 above)
Comment (ilchu) [inline config/src/client.rs:155]:
```suggestion
                    "`TORII_API_URL` string: `{api_url}` only supports the `HTTP` protocol currently".to_owned()
```

**Review by Erigara — APPROVED** [review]

File: config/src/iroha.rs [hunk 3]
```diff
@@ -120,8 +120,9 @@ impl ConfigurationProxy {
                         ),
                     ));
                 } else {
+                    // TODO: should we just warn the user that this value will be ignored?
```
Comment (ilchu) [inline config/src/iroha.rs:123]: Or rather, fail only if it's not equivalent to the one provided in upmost config? That could be a valid option for the above failure points too.

File: config/src/client.rs [hunk 4]
```diff
@@ -32,7 +32,7 @@ impl FromStr for WebLogin {
     type Err = eyre::ErrReport;
     fn from_str(login: &str) -> Result<Self> {
         if login.contains(':') {
-            return Err(eyre!("WebLogin cannot contain the `:` character"));
+            eyre::bail!("The `:` character, in `{login}` is not allowed");
```
Comment (outoftardis) [inline config/src/client.rs:35]:
```suggestion
            eyre::bail!("The `:` character in `{login}` is not allowed");
```

File: config/src/client.rs [hunk 5]
```diff
@@ -119,35 +123,36 @@ impl ConfigurationProxy {
...
                 eyre::bail!(
-                    ConfigError::ProxyBuildError("Really small `TRANSACTION_TIME_TO_LIVE_MS`, network throughput may be compromised".to_owned())
+                    ConfigError::ProxyBuildError("`TRANSACTION_TIME_TO_LIVE_MS`, network throughput may be compromised for values less than {TTL_TOO_SMALL_THRESHOLD}".to_owned())
                 );
             }
             // Timeouts bigger than transaction TTL don't make sense as then transaction would be discarded before this timeout
             if let Some(timeout) = self.transaction_status_timeout_ms {
                 if timeout > tx_ttl {
-                    eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_STATUS_TIMEOUT_MS` bigger than `TRANSACTION_TIME_TO_LIVE_MS`, consider making it smaller".to_owned()));
+                    eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_STATUS_TIMEOUT_MS`: {timeout} bigger than `TRANSACTION_TIME_TO_LIVE_MS`: {self.transaction_status_timeout_ms}, consider making it smaller".to_owned()));
```
Comment (outoftardis) [inline config/src/client.rs:134]:
```suggestion
                    eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_STATUS_TIMEOUT_MS`: {timeout} bigger than `TRANSACTION_TIME_TO_LIVE_MS`: {self.transaction_status_timeout_ms}. Consider making it smaller".to_owned()));
```

File: config/src/client.rs [hunk 6]
```diff
@@ -119,35 +123,36 @@ impl ConfigurationProxy {
...
                 }
             }
         }
         if let Some(tx_limits) = self.transaction_limits {
-            // 1 KiB
-            if tx_limits.max_wasm_size_bytes < 2_u64.pow(10) {
-                eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_LIMITS` parameter's `max_wasm_size` field really small, consider making it bigger".to_owned()));
+            if tx_limits.max_wasm_size_bytes < WASM_SIZE_TOO_SMALL_THRESHOLD {
+                eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_LIMITS` parameter's `max_wasm_size` field too small at {tx_limits.max_wasm_size_bytes}, consider making it bigger than {WASM_SIZE_TOO_SMALL_THRESHOLD}".to_owned()));
```
Comment (outoftardis) [inline config/src/client.rs:140]:
```suggestion
                eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_LIMITS` parameter's `max_wasm_size` field too small at {tx_limits.max_wasm_size_bytes}. Consider making it bigger than {WASM_SIZE_TOO_SMALL_THRESHOLD}".to_owned()));
```

**Review by ilchu — APPROVED** [review]

**Pull request merged**